### ISSUE TITLE
Provide a tutorial and general tooling scripts for YARN and launching client applications

### ISSUE THREAD

**Issue by kkasravi** [intel-hadoop/gearpump#762]
General scripts to launch the GearPump application master and GearPump applications that run on the application master need to be added. A short tutorial on how to launch the pipeline application is required.

**Comment by kkasravi** [comment]
Some scripts I've developed that have made launching GearPump on YARN and submitting an application to the GearPump master much easier

gp -h
usage: ./gp options commands
OPTIONS:
  -h      Show this message
COMMANDS:
  show:     show the application master address
  start:    starts the application master first killing any active application master 
  stop:     kills any active application master
  pipeline: launches the pipeline application. Will first check if an application master is active

runexample -h
usage: ./runexample options <appname>
OPTIONS:
   -h      Show this message
   -m      Master
   -p      Prefix
   -v      Verbose

**Comment by kkasravi** [comment]
Based on Sean's idea of having a fat jar per example and invoking it like:
```
gear app –jar example-wordcount.jar
gear app –jar example-sol.jar
gear app –jar example-pipeline.jar
```
We may be able to just shorten this to something like
```
gear app wordcount
```
For examples where more than one main class is bundled in the assembly we could have an additional class short name 
```
gear app kafka KafkaWordCount
```
The algorithm would be something like:
1. Get version by reading version.sbt
2. Derive assembly name based on wordcount  (output/target/pack/examples/gearpump-examples-assembly-${APP}-${VERSION}.jar)
3. Verify existence of this jar and class load it.
4. Derive package name based on wordcount (org.apache.gearpump.examples.wordcount)
5. Using reflection look for all classes that extend App under this package

The runexample script sort of does this now – which could be converted to scala and moved into AppSubmitter
From runexample: 
```
#!/bin/bash
…
case "$1" in
  complexdag)
    JAR=output/target/pack/examples/gearpump-examples-assembly-${VERSION}.jar
    MAIN=org.apache.gearpump.streaming.examples.${1}.Dag
    ;;
  wordcount)
    JAR=output/target/pack/examples/gearpump-examples-assembly-${VERSION}.jar
    MAIN=org.apache.gearpump.streaming.examples.${1}.WordCount
    ;;
  fsio)
    JAR=output/target/pack/examples/gearpump-examples-assembly-${VERSION}.jar
    MAIN=org.apache.gearpump.streaming.examples.${1}.SequenceFileIO
    ;;
  pipeline)
    JAR=experiments/kafka-hbase-${1}/target/scala-2.11/gearpump-experiments-kafka-hbase-${1}-assembly-${VERSION}.jar
    MAIN=org.apache.gearpump.experiments.pipeline.PipeLine
    ARGS="-conf conf/pipeline.conf"
    ;;
  sol)
    JAR=output/target/pack/examples/gearpump-examples-assembly-${VERSION}.jar
    MAIN=org.apache.gearpump.streaming.examples.${1}.SOL
    ;;
  kafkawordcount)
    JAR=output/target/pack/examples/gearpump-examples-assembly-${VERSION}.jar
    MAIN=org.apache.gearpump.streaming.examples.kafka.wordcount.KafkaWordCount
    ;;
  ?)
    usage
    exit
    ;;
```

**Comment by clockfly** [comment]
I don’t  we should try to infer anything(like package name, or jar path) implicitly, it is just too tricky, and they cannot be actually be inferred, considering a user custom jar,instead of built-in ones.
 
How about this,
If user have specified a main class, then we will use the old way to start the app
gear app –jar xx.jar gearpump-wordcount.jar xx.MainClass 
 
If user have not specified a main class, 
Gearpump app –jar gearpump-wordcount.jar 
Then we will try to resolve the MainClass definition in MANIFEST of gearpump-wordcount.jar. If we find nothing, then we throw exception.
 
Current there is a version name appended in the jar file name, to make user easier to submit a job, we can create a shortcut for example jars.
For example: example-wordcount.jar (short link) linkto example-wordcount-0.3.3-SNAPSHOT.jar

**Comment by clockfly** [comment]
If we cannot make the shortcut working on windows, then we will not add the short cut feature.

**Comment by clockfly** [comment]
Not updated for long time. Close this.
